src/Simplify.elm

```diff
@@ -8872,89 +8872,70 @@ needsParens expr =
 
 isAlwaysMaybe : ModuleNameLookupTable -> Node Expression -> Match (Maybe { fixes : List Fix, throughLambdaFunction : Bool })
 isAlwaysMaybe lookupTable baseExpressionNode =
-    let
-        expressionWithoutParensNode : Node Expression
-        expressionWithoutParensNode =
-            AstHelpers.removeParens baseExpressionNode
-    in
-    case Node.value expressionWithoutParensNode of
-        Expression.FunctionOrValue _ "Just" ->
-            case ModuleNameLookupTable.moduleNameFor lookupTable expressionWithoutParensNode of
-                Just [ "Maybe" ] ->
-                    Determined
-                        (Just
-                            { fixes = [ Fix.removeRange (Node.range expressionWithoutParensNode) ]
-                            , throughLambdaFunction = False
-                            }
-                        )
+    case AstHelpers.getSpecificValueOrFunction ( [ "Result" ], "Err" ) lookupTable baseExpressionNode of
```

Comment:
Comment on commit description: If I see a commit saying "Rename X to Y", I will tend to believe that you *only* did that change. And because in Elm renaming things is very safe, I will in practice skip reviewing that commit. I'll at most look at the old name and the new name. I'll try to use the same logic when I need to understand in which commit a bug was introduced to find the commit faster.

Here, it looks like you did other things though, which is confusing. I highly recommend that when a task is splittable, that you do it in 2 commits.

Another example of where splitting things in 2 commits would make reviewing a commit much easier is when you introduce `firstThatReportsError` in a function. Refactoring a function to use it is pretty simple and predictable, to the point where if you have a commit saying "Make X use firstThatReportsError" I would skim over it. And then have a separate commit to introduce different changes like using a different AstHelpers function. Combining two two, with the added indentation makes it really much harder to see what has or has not changed.

On the topic, I really enjoyed [this talk](https://www.youtube.com/watch?v=wp6oSVDdbXQ&pp=ygUPbGxld2VsbHluIGZhbGNv) :relaxed: